src/processes/linux_list.rs

```diff
@@ -0,0 +1,51 @@
+use crate::processes::{ProcessInfo, ProcessList};
+use anyhow::Result;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use sysinfo::{ProcessRefreshKind, ProcessesToUpdate, System, UpdateKind};
+
+pub fn active_executables() -> Result<ProcessList> {
+    let mut executables: HashMap<PathBuf, ProcessInfo> = HashMap::new();
+    let mut sys = System::new();
+
+    sys.refresh_processes_specifics(
+        ProcessesToUpdate::All,
+        true,
+        ProcessRefreshKind::nothing().with_exe(UpdateKind::OnlyIfNotSet),
+    );
+
+    for process in sys.processes().values() {
+        // process.exe() will return empty path if there was an error while trying to read /proc/<pid>/exe.
+        if let Some(exec) = process.exe() {
+            let exec_buf = exec.to_path_buf();
+
+            match executables.entry(exec_buf) {
+                Entry::Occupied(_) => {}
+                Entry::Vacant(e) => {
+                    let exec_buf = e.key().clone();
+                    // process display name can contain non-UTF-8 characters, forcing us to use to_string_lossy
+                    let display_name = process.name().to_string_lossy().to_string();
```

Comment:
What kind of values are we getting for `name`? Is this actually better than reusing `process.exe()`, or is it just the same but cut off?

Reply:
`process.exe()` gives us the absolute path to the executable (eg. `/usr/bin/python3.11`) while `process.name()` gives us the display name for the process (eg. `python3`). I think `display_name` and `executable` should be different because it could get a bit clumsy using the executable path to represent a process. [macos_list.rs](https://github.com/mitmproxy/mitmproxy_rs/blob/49e78d6859781755b602337671eaa4cad269a5b8/src/processes/macos_list.rs#L27) follows a similar approach.